geetools/ee_string.py

```diff
@@ -55,17 +59,32 @@ def format(self, template: dict | ee.Dictionary) -> ee.String:
 
                 initialize_documentation()
 
-                s = ee.String("{greeting} {name} !")
-                s = s.geetools.format({"greeting": "Hello", "name": "bob"})
+                s = ee.String("{greeting} developer, pi={number%.2f} start={start%tyyyy-MM-dd} end={end%tdd MMM yyyy}")
+                s = s.geetools.format({"greeting": "Hello", "number": 3.1415, "start": 1577836800000, "end": "2021-01-01"})
                 s.getInfo()
         """
         template = ee.Dictionary(template)
-        templateList = template.keys().zip(template.values())
-
-        def replace_format(kv, s):
-            kv = ee.List(kv)
-            key, value = ee.String(kv.get(0)), ee.String(kv.get(1))
-            pattern = ee.String("{").cat(key).cat(ee.String("}"))
-            return ee.String(s).replace(pattern, value)
-
-        return ee.String(templateList.iterate(replace_format, self._obj))
+        template.keys().zip(template.values())
+
+        toFormat = self._obj.match("\\{([^\\}]+)\\}", "g")
+
+        def replace_format(string, init):
+            clean = ee.String(string).slice(1, -1)
+            parts = clean.split("%")
+            keys = ee.List.sequence(1, parts.size()).map(lambda k: ee.Number(k).toInt().format())
+            rel = ee.Dictionary.fromLists(keys, parts)
+            key = ee.String(rel.get("1"))
+            value = template.get(key, "")
+            format = ee.String(rel.get("2", ""))
+            is_date = format.match("^t.+").size().gt(0)
```

Comment:
this is a key piece of information that i didn't catched on my first try, you need to add "%t" to notify a time and a "%" for numbers right ? if yes can you add this information in the docstring in plain english if not this will never be used.

Reply:
exactly, and you're right, I should improve the documentation

Reply:
@12rambau docstring updated